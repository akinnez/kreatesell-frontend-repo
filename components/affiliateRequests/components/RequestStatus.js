import Tags from "components/Tags";

const RequestStatus = ({ status }) => (
  <>
    {status === "Approved" ? (
      <Tags color="green">{status}</Tags>
    ) : status === "Declined" ? (
      <Tags color="red">{status}</Tags>
    ) : status === "Pending" ? (
      <Tags color="orange">{status}</Tags>
    ) : (
      <Tags>{status}</Tags>
    )}
  </>
);

export default RequestStatus;
