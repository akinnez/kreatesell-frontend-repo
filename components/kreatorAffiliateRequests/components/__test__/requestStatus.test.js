import RequestStatus from "../RequestStatus";
import renderer from "react-test-renderer";

describe("RequestStatus : ", () => {
  it("renders a snapshot of <RequestStatus /> when status is Approved", () => {
    const status = renderer
      .create(<RequestStatus status="Approved" />)
      .toJSON();
    expect(status).toMatchSnapshot();
  });
  it("renders a snapshot of <RequestStatus /> when status is Pending ", () => {
    const status = renderer.create(<RequestStatus status="Pending" />).toJSON();
    expect(status).toMatchSnapshot();
  });
  it("renders a snapshot of <RequestStatus /> when status is Declined ", () => {
    const status = renderer
      .create(<RequestStatus status="Declined" />)
      .toJSON();
    expect(status).toMatchSnapshot();
  });
});
