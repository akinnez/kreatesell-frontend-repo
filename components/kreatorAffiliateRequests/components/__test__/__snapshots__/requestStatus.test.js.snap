// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestStatus :  renders a snapshot of <RequestStatus /> when status is Approved 1`] = `
<span
  className="tag green"
>
  Approved
</span>
`;

exports[`RequestStatus :  renders a snapshot of <RequestStatus /> when status is Declined  1`] = `
<span
  className="tag red"
>
  Declined
</span>
`;

exports[`RequestStatus :  renders a snapshot of <RequestStatus /> when status is Pending  1`] = `
<span
  className="tag orange"
>
  Pending
</span>
`;
